### PR TITLE
Don't use die() but echo to skip tests.

### PR DESCRIPTION
**Pull request by staabm**
To unlock PHPUnit side-effect-free in-process `--SKIPIF--` evaluation, we use regular `echo` instead of `die()`.
That way PHPunit can detect whether conditions may be evaluated in the main-process to reduce sub-process creation overhead.

Depends on the work-inprogress of https://github.com/sebastianbergmann/phpunit/pull/5974

----

Running phpunit with the side-effect detecting PR vs. the regular master version, yields the following results in this PR on my windows laptop:

```
$ hyperfine '/c/tools/php83/php ../phpunit/phpunit tests' '/c/tools/php83/php vendor/bin/phpunit tests' --warmup=3
Benchmark 1: C:/tools/php83/php ../phpunit/phpunit tests
  Time (mean ± σ):      3.312 s ±  0.020 s    [User: 0.184 s, System: 0.510 s]
  Range (min … max):    3.291 s …  3.347 s    10 runs

Benchmark 2: C:/tools/php83/php vendor/bin/phpunit tests
  Time (mean ± σ):      3.555 s ±  0.031 s    [User: 0.220 s, System: 0.599 s]
  Range (min … max):    3.521 s …  3.604 s    10 runs

Summary
  C:/tools/php83/php ../phpunit/phpunit tests ran
    1.07 ± 0.01 times faster than C:/tools/php83/php vendor/bin/phpunit tests

$ /c/tools/php83/php -v
PHP 8.3.11 (cli) (built: Aug 27 2024 21:28:35) (NTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.3.11, Copyright (c) Zend Technologies
```

-> inline-skip-if evaluation of 7 test-cases - as proposed with this PR - will yield a 7-8% perf improvement.

### CODE REVIEW

File: tests/never-typed.phpt [hunk 1]
```diff
@@ -2,9 +2,8 @@
 https://github.com/antecedent/patchwork/issues/140
 
 --SKIPIF--
-<?php version_compare(PHP_VERSION, "8.1", ">=")
-      or die("skip because this bug only occurs in PHP 8.1") ?>
-
+<?php
+if (!version_compare(PHP_VERSION, "8.1", ">=")) echo "skip because attributes are only available since PHP 8.1";
```
Comment (anomiex) [inline tests/never-typed.phpt:6]: Why did you change the message here?

Comment (staabm) [reply tests/never-typed.phpt]: good catch. copy paste mistake

**Comment by jrfnl** [comment]
Out of curiousity: how does PHPUnit decide whether the test should be skipped or not ?
I thought that was related to the exit code of the script run in the `SKIPIF` process ?

If that's no longer the case, the criteria for how the test skipping is determined in `phpt` tests need to be clearly outlined and documented.

File: tests/anonymous-class-attribute.phpt [hunk 2]
```diff
@@ -2,9 +2,8 @@
 Attribute declared in anonymous class
 
 --SKIPIF--
-<?php version_compare(PHP_VERSION, "8.0", ">=")
-      or die("skip because attributes are only available since PHP 8.0") ?>
-
+<?php
+if (!version_compare(PHP_VERSION, "8.0", ">=")) echo "skip because attributes are only available since PHP 8.0";
```
Comment (jrfnl) [inline tests/anonymous-class-attribute.phpt:6]:
```suggestion
if (!version_compare(PHP_VERSION, "8.0", ">=")) {
    echo "skip because attributes are only available since PHP 8.0";
}
```

As the `SKIPIF` is no longer a conditional expression with `or`, might be good to use a little cleaner code style for the code. (here and elsewhere)

**Comment by staabm** [comment]
> Out of curiousity: how does PHPUnit decide whether the test should be skipped or not ?
> I thought that was related to the exit code of the script run in the `SKIPIF` process ?

in the current stable release a test is skipped, when the executed code of the `SKIPIF` section contains the word `skip`
see https://github.com/sebastianbergmann/phpunit/blob/a82cc054802a11b786f3159030a93d106335cb4d/src/Runner/PHPT/PhptTestCase.php#L420-L451

**Comment by staabm** [comment]
a similar change was merged today into symfony with https://github.com/symfony/symfony/pull/58680

**Review by jrfnl — APPROVED** [review]
LGTM.

While the performance improvement will only be seen on PHPUnit 11.5, this change doesn't seem to negatively impact runs on older PHPUnit versions, so I see no reason not to merge this.

I also think it would be good to document this feature for PHPT tests, but that's a task for the PHPUnit website (which currently doesn't have any documentation on PHPT tests - see https://github.com/sebastianbergmann/phpunit-documentation-english/issues/302 ).
